flask/app/models.py

```diff
@@ -106,7 +111,14 @@ class ParticipantType(str, Enum):
 class Participant(db.Model):
     participant_id: int = db.Column(db.Integer, primary_key=True)
     family_id = db.Column(db.Integer, db.ForeignKey("family.family_id"), nullable=False)
-    participant_codename: str = db.Column(db.String(50), nullable=False, unique=True)
+    participant_codename: str = db.Column(
+        db.String(50),
+        CheckConstraint(
+            "participant_codename REGEXP '^[.,[:alnum:],-,_]+$' AND participant_codename NOT REGEXP '^[.]+$'"
```

Comment:
Need to escape the minus sign in order to match for it in a bracket expression. Also unless we're matching for commas, the characters don't need to be comma-delimited.
```suggestion
            r"participant_codename REGEXP '^[.[:alnum:]\-_]+$' AND participant_codename NOT REGEXP '^[.]+$'"
```